pulp_smash/tests/pulp2/rpm/api_v2/test_errata.py

```diff
@@ -0,0 +1,87 @@
+"""Test whether applying an ``Errata`` will update an RPM package version."""
+import unittest
+from urllib.parse import urljoin
+
+from pulp_smash import api, cli, config, utils
+from pulp_smash.constants import REPOSITORY_PATH, RPM_UNSIGNED_FEED_URL
+from pulp_smash.tests.pulp2.rpm.api_v2.utils import gen_distributor, gen_repo
+from pulp_smash.tests.pulp2.rpm.utils import (
+    gen_yum_config_file,
+    get_rpm_names_versions,
+)
+from pulp_smash.tests.pulp2.rpm.utils import set_up_module as setUpModule  # noqa pylint:disable=unused-import
+
+
+class ErrataLatestPackageVersionTestCase(unittest.TestCase):
+    """Test whether applying an Errata will update a RPM package version."""
+
+    def test_all(self):
+        """Test whether applying an Errata will update a RPM package version.
+
+        This test targets the following issue:
+
+        * `Pulp Smash #760 <https://github.com/PulpQE/pulp-smash/issues/760>`_
+
+        Do the following:
+
+        1. Create, sync and publish a repository with ``Errata``, and RPM
+           packages.
+           Two of these RPMs have the **same name**, and **different
+           versions**.
+           The newest version is part of the ``Errata``.
+        2. Install the oldest version of the aforementioned RPM.
+        3. Apply the ``Errata``, and verify that latest version of previous
+           installed package was updated properly.
+        """
+        # Create, sync and publish a repository.
+        cfg = config.get_config()
+        client = api.Client(cfg, api.json_handler)
+        pkg_mgr = cli.PackageManager(cfg)
+        sudo = '' if utils.is_root(cfg) else 'sudo '
+        verify = cfg.get_systems('api')[0].roles['api'].get('verify')
+        body = gen_repo()
+        body['importer_config']['feed'] = RPM_UNSIGNED_FEED_URL
+        body['distributors'] = [gen_distributor()]
+        repo = client.post(REPOSITORY_PATH, body)
+        self.addCleanup(client.delete, repo['_href'])
+        repo = client.get(repo['_href'], params={'details': True})
+        utils.sync_repo(cfg, repo)
+        utils.publish_repo(cfg, repo)
+
+        # Pick a RPM with two versions.
+        rpm_name = 'walrus'
+        rpm_versions = get_rpm_names_versions(cfg, repo['id'])[rpm_name]
+        cli_client = cli.Client(
+            cfg,
+            cli.code_handler,
+            pulp_system=cfg.get_systems('shell')[0]
```

Comment:
By default, a CLI client talks to the first system that is declared as fulfilling the `pulp cli` role. That is, by default, the CLI clients talks to the first system that is declared as having `pulp-admin` available for use. By definition, a system that fulfills the `pulp cli` role must also fulfill the `shell` role. In this case, I think it's OK to omit the `pulp_system` argument.